src/Native/Unix/System.Native/pal_termios.cpp

```diff
@@ -0,0 +1,356 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+#include "pal_types.h"
+#include <termios.h>
+#include <sys/ioctl.h>
+#include <stdio.h>
+
+/* This is dup of System/IO/Ports/NativeMethods.cs */
+enum {
+    NoneParity = 0,
+    OddParity = 1,
+    EvenParity = 2,
+    MarkParity = 3,
+    SpaceParity = 4
+};
+
+/* System/IO/Ports/Handshake.cs */
+enum {
+    HandshakeNone,          // No Flow Control
+    HandshakeSoft,          // Software Flow Control -> XOn/Xoff
+    HandshakeHard,          // Hardware Flow Control -> RTS/CTS
+    HandshakeBoth           // Software & Hardware Flow Control
+};
+
+static int  SystemNative_Termios_GetStatus(int fd)
+{
+    int status = 0;
+    if(ioctl(fd, TIOCMGET, &status) < 0)
+    {
+        return -1;
+    }
+    return status;
+}
+
+static int  SystemNative_Termios_SetStatus(int fd, int status)
+{
+    if (ioctl(fd, TIOCMSET, &status) < 0)
```

Comment:
In this case it seems that we don't need the if statement it can just be `return ioctl(fd, TIOCMSET, &status);` at least per http://man7.org/linux/man-pages/man4/tty_ioctl.4.html